### DISCUSSION TITLE
Restore of SealedSecrets and Secrets clashes

### DISCUSSION THREAD

**Discussion by BenjaminHerbert** [vmware-tanzu/velero#3848] [Community support Q&A]
Hi there,

I am running [Bitnami's sealed-secret controller](https://github.com/bitnami-labs/sealed-secrets) in a cluster to manage some of my secrets.
It works as follows: one creates a SealedSecret using the controller's private key and the controller decrypts this into a secret.

While restoring a backup for a migration, I ran into the issue that Secrets are restored before the "corresponding" SealedSecret and the owner relations are "lost". This causes the SealedSecret-Controller to fail, as the "secret" is already present. I have to manually parse the log of the controller and delete the obstructing Secrets.

Is there a way only backup Secrets that do not have an "owner" reference? Any other ideas?

Best regards
Benjamin

**Comment by sseago** [comment]
I wonder whether writing a restore plugin might resolve this in your situation. Does the SealedSecret contain everything needed to recreate the Secret? If so, then you might write a velero Secret RestoreItemAction plugin which discards any secret with a SealedSecret owner reference. Also, maybe there's a way to propose a change to the SealedSecret controller that allows it to "adopt" the secret if it already exists (and has the proper content), although I've never used SealedSecrets before, so I don't know whether this is possible. Seems like the controler failing when the secret exists might be a more general problem with backup/restore and that controller, so fixing the issue there would make it work with velero.
